lints/etsi/lint_qcstatem_psd2_policy_recommended.go

```diff
@@ -0,0 +1,51 @@
+package etsi
+
+/*
+ * ZLint Copyright 2020 Regents of the University of Michigan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import (
+	"github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zlint/lint"
+	"github.com/zmap/zlint/util"
+)
+
+type qcStatemPsd2PolicyRecomm struct{}
+
+func (l *qcStatemPsd2PolicyRecomm) Initialize() error {
+	return nil
+}
+
+func (l *qcStatemPsd2PolicyRecomm) CheckApplies(c *x509.Certificate) bool {
+	isPresent := util.HasCertAnyEtsiQcStatement(c)
+	return isPresent
+}
+
+func (l *qcStatemPsd2PolicyRecomm) Execute(c *x509.Certificate) *lint.LintResult {
+	if util.HasCertAnyEtsiQcStatement(c) && !util.HasCertAnyEtsiQcpPolicy(c) {
```

Comment:
There's a challenge here, because the ETSI QC Statements defined in 319 412-5 do not necessarily imply a certificate is a Qualified Certificate. 412-5 makes that clear in [Section 1: Scope](https://www.etsi.org/deliver/etsi_en/319400_319499/31941205/02.02.01_60/en_31941205v020201p.pdf#page=6) that they can be used in combination with any certificate profile.

That is:
- If (QCP Present), then (QC Statement MUST be present)
- If (QC Statement present), then (QCP MAY be present)

However, this is even messier, because 412-4 Section 4.3 only states that the policies from 5.3 "SHOULD" be present, not MUST. This is messy, because GEN-6.6.1-05 of 319 411-2 allows a TSP-specific policy OID to be used in lieu of the 5.3 OIDs.

So I'm not even sure this makes sense as a warning, because the SHOULD is effectively weakened to a MAY by GEN-6.6.1-05.

So I don't think we want this as a lint, but what we do want is that if they assert any of the OIDs from 5.3 of ETSI EN 319 411-2, then they're bound to include QCStatements by GEN-6.6.1-02 (although whether or not that is a hard requirement is iffy, given that only 'appropriate' QCStatements are required). Additionally, an approach like https://github.com/zmap/zlint/blob/fd40f579253ea1ebfb18a585ab5cd8e7dcde61aa/v2/util/ev.go#L21 would be needed to track the qualified certificate profiles that are CA specific.

Does that all make sense?